src/commands/build-util.ts

```diff
@@ -1,19 +1,112 @@
 import type CSSModuleLoader from 'css-modules-loader-core';
-import type {EventEmitter} from 'events';
-import execa from 'execa';
-import npmRunPath from 'npm-run-path';
-import {
-  BuildScript,
-  SnowpackConfig,
-  SnowpackPluginBuildArgs,
-  SnowpackPluginBuildResult,
-} from '../config';
+import {EventEmitter} from 'events';
+import path from 'path';
+import {SnowpackBuildMap, SnowpackConfig, SnowpackPlugin, SnowpackSourceFile} from '../config';
+import {getExt, replaceExt} from '../util';
+
+export interface BuildFileOptions {
+  buildPipeline: SnowpackPlugin[];
+  messageBus: EventEmitter;
+  isDev: boolean;
+}
 
 const IS_PREACT = /from\s+['"]preact['"]/;
 export function checkIsPreact(filePath: string, contents: string) {
   return filePath.endsWith('.jsx') && IS_PREACT.test(contents);
 }
 
+export function getInputsFromOutput(fileLoc: string, plugins: SnowpackPlugin[]) {
+  const ext = path.extname(fileLoc);
+  const potentialInputs = new Set([fileLoc]);
+  for (const plugin of plugins) {
+    if (plugin.output.includes(ext)) {
+      plugin.input.forEach((inp) => potentialInputs.add(fileLoc.replace(ext, inp)));
+    }
+  }
+  return Array.from(potentialInputs);
+}
```

Comment:
Just a note: though usually I’m -1 on unit tests, these mapping functions were the one thing where I wanted to test lots of scenarios quickly. Thoughts on this (doesn’t have to be in this PR, just eventually)?

Reply:
Definitely +1 on that, now that there's some lower-level structure and reuse it's becoming much more clear where unit tests can help us